Be more explicit about future changes to `mode` (#231)
ipfs/specs: UNIXFS.md

UnixFS currently supports two optional metadata fields:

* `mode` -- The `mode` is for persisting the file permissions in [numeric notation](https://en.wikipedia.org/wiki/File_system_permissions#Numeric_notation) \[[spec](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_stat.h.html)\].
  - If unspecified this defaults to
    - `0755` for directories/HAMT shards
    - `0644` for all other types where applicable
  - The nine least significant bits represent  `ugo-rwx`
  - The next three least significant bits represent `setuid`, `setgid` and the `sticky bit`
  - The remaining 20 bits are reserved for future use, and are subject to change. Spec implementations **MUST** handle bits they do not expect as follows:

>     - For future-proofing the (de)serialization layer must preserve the entre uint32 value during clone/copy operations, modifying only bit values that have a well defined meaning: `clonedValue = ( modifiedBits & 07777 ) | ( originalValue & 0xFFFFF000 )`

Reviewer: ```suggestion
    - For future-proofing the (de)serialization layer must preserve the entire uint32 value during clone/copy operations, modifying only bit values that have a well defined meaning: `clonedValue = ( modifiedBits & 07777 ) | ( originalValue & 0xFFFFF000 )`
```

>     - Any higher level operations interpreting the value must proactively mask-off bits without a defined meaning in the current version of the spec: `interpretedValue = originalValue & 07777`

Reviewer: Implementations of this spec should handle this, otherwise the [abstraction leaks](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/).

I think it's fair that code written against a given implementation is expecting it to implement a certain version of the spec, so the implementation should be responsible for masking and/or otherwise making sure it only allows access to values it expects, not the calling code.

You might choose to do this if you need to code defensively, but interacting with low level details like this should not be a requirement of use for high level components.

```suggestion
    - Implementations of this spec must proactively mask off bits without a defined meaning in the implemented version of the spec: `interpretedValue = originalValue & 07777`
```

Reviewer: All looks good, otherwise.
